src/python/cli_new/lib/cli/config.py

```diff
@@ -119,6 +120,94 @@ def master(self):
 
         return master
 
+    def principal(self):
+        """
+        Return the principal in the configuration file
+        """
+        principal = self.data["master"].get("principal")
+        if principal is None:
+            return None
+        elif not principal:
+            raise CLIException("The 'principal' field in 'master'"
+                               " must be non-empty")
+
+        return principal
+
+    def secret(self):
+        """
+        Return the secret in the configuration file
+        """
+        secret = self.data["master"].get("secret")
+        if secret is None:
+            return None
+        elif not secret:
+            raise CLIException("The 'secret' field in 'master'"
+                               " must be non-empty")
+
+        return secret
+
+    def agent_ssl(self):
+        """
+        Return if the agent support ssl
+        """
+        if "agent" in self.data:
+            agent_ssl = self.data["agent"].get("ssl")
+            if agent_ssl is None:
+                return None
+            elif not agent_ssl:
+                if not isinstance(self.data["agent"]["ssl"], bool):
```

Comment:
```
isinstance(agent_ssl, bool)
```

Reply:
Is already change in the current version.  But I'm interesting of what do u think about to change `def secret` and `def principal` to 

```
    def secret(self):
        return self.data["master"].get("secret")
```
No matter secret does not exist, or it's empty, secret would be "None". Thats why the "elif" is not usefull in these case. And because secret does not have to be configured, it's also not important to rise a "Exception".  What do u think?

Reply:
> Is already change in the current version.

I don't seem to see it.

Also, I still see 3 commits, I thought you'd squashed?

Reply:
> But I'm interesting of what do u think about to change `def secret` and `def principal` to
> 
> ```
>     def secret(self):
>         return self.data["master"].get("secret")
> ```
> 
> No matter secret does not exist, or it's empty, secret would be "None". Thats why the "elif" is not usefull in these case. And because secret does not have to be configured, it's also not important to rise a "Exception". What do u think?

Yes, if it's not required it makes sense.

Reply:
:+1:

Reply:
Yeah I didnt commit and pushed just now. I wanted to wait until I get the feedback to the "def secret". :-)  Then I will also squash again.